liquidhaskell-boot/src/Language/Haskell/Liquid/Types/Specs.hs

```diff
@@ -757,6 +760,8 @@ data LiftedSpec = LiftedSpec
   , liftedBounds     :: RRBEnvV LHName LocBareTypeLHName
   , liftedAxeqs      :: HashSet (F.EquationV LHName)
     -- ^ Equalities used for Proof-By-Evaluation
+  , liftedDefines    :: M.HashMap F.Symbol LMap
```

Comment:
Looks like both `defines` and `liftedDefines` could use `LMap`.

If this is to be aligned with the latest changes to name resolution, LMap should be generalized to something like
```Haskell
type LMap = LMapV Symbol
data LMapV v = LMap
  { lmVar  :: F.LocSymbol
  , lmArgs :: [Symbol]
  , lmExpr :: ExprV v
  }
```
and `liftedDefines` should be defined as `M.HashMap F.Symbol (LMapV LHName)`. This ensures names in the rhs are resolved by the time the specs are serialized.